Restructure documentation

PhilippeMoussalli opened the pull request:
PR that restructures the documentation to take in mind the user's journey #572 

Includes other changes:
* Updating main README documentation (components, some text tweaks, runner visuals, ...)
* Renaming some components to match the format <component_function>_<modality> (e.g. text_normalization -> normalize_text) 

 There are many other changes that need to be made but best to track them in separate PRs to make reviewing them easier. Created issues can be found here #572

RobbeSneyders commented on README.md:
```diff
@@ -24,50 +24,66 @@ It offers:
 </li>
 <li>
-🧱 Library of off-the-shelf reusable components for
+🧱 Off-the-shelf library of easily adaptable reusable components for:
```
Nit, but I don't think "off-the-shelf" is used correctly here. The components are off-the-shelf, not the library :slightly_smiling_face:

RobbeSneyders commented on README.md:
```diff
@@ -24,50 +24,66 @@ It offers:
+    🔧 Plug ‘n’ play composable pipelines for creating datasets for:
     <ul>
         <li>AI image generation model fine-tuning (Stable Diffusion, ControlNet)</li>
         <li>Large language model fine-tuning (LLaMA, Falcon)</li>
```
Let's add RAG here already as well.

PhilippeMoussalli replied:
Done :)

RobbeSneyders commented on README.md:
```diff
@@ -24,50 +24,66 @@ It offers:
+    Data Filtering
     <ul>
-        <li>Content, e.g. language, visual style, topic, format, aesthetics, etc.</li>
-        <li>Context, e.g. copyright license, origin</li>
-        <li>Metadata</li>
+        <li>Content: language, visual style, topic, format, aesthetics, NSFW, etc.</li>
+        <li>Context: e.g. copyright license, origin, etc.</li>
+        <li>Deduplication</li>
+    </ul>
+    </li>
+    <li>
+    Data Enrichment
+    <ul>
+        <li>Captions</li>
+        <li>Segmentations</li>
+        <li>Embeddings</li>
+    </ul>
+    </li>
+    <li>
+    Data Transformation
+    <ul>
+        <li>Images: cropping, resizing, etc. </li>
+        <li>Text: chunking, indexing, etc. </li>
+    </ul>
+    </li>
+    <li>
+    Data retrieval
+    <ul>
+        <li>Common Crawl</li>
+        <li>LAION</li>
```
Can we summarize every category of filtering on a single line? Just so the view is a bit more condensed.

RobbeSneyders commented on README.md:
```diff
@@ -24,50 +24,66 @@ It offers:
-<li>🏭 Production-ready, scalable deployment</li>
-<li>☁️ Multi-cloud integrations</li>
+<li>🐍 A simple Pandas based interface for creating custom components</li>
+<li>🏭 Scalable deployment for production environments.</li>
```
I would keep `production-ready` here, as it's more of a USP of Fondant than `scalable`.

RobbeSneyders commented on README.md:
```diff
@@ -24,50 +24,66 @@ It offers:
-machine learning-based operations. This need not be the case, however, if processing
-components would be reusable and exchangeable and pipelines were easily composable.
-Realizing this is the main vision behind Fondant.
+In the era of Foundation Models, preserving data control has become paramount. 
+Yet, the construction of pipelines for extensive data processing remains a costly affair, particularly in the context of 
+complex machine learning-based operations. 
+Fondant's mission is to streamline this process by introducing reusable and interchangeable processing components, 
+thus establishing easily composed pipelines as the new norm.
```
I'm not the biggest fan of this rewrite, I think the new text is more complex.

RobbeSneyders commented on README.md:
```diff
@@ -153,22 +178,39 @@ pip install git+https://github.com/ml6team/fondant.git
+            </figure>
+        </td>
+    </tr>
+</table>
```
I would already add a Sagemaker Runner as well, with some :construction: Under construction :construction:  warning :)

PhilippeMoussalli replied:
Good point, done ;)

RobbeSneyders commented on docs/caching.md:
We start with a very technical explanation of how caching is implemented. Can we start with a high level explanation of why, how, and when to use it?

PhilippeMoussalli replied:
Created a ticket for this #610

RobbeSneyders commented on docs/documentation_guide.md:
```diff
@@ -0,0 +1,95 @@
+to/from Fondant.  
+-> Build your own [custom component](components/custom_component.md) using the Fondant component
+API.
+-> Learn how to publish your own [components](components/publishing_components.md) to a container
```
I think you need two spaces at the end of the line above to make this one appear on a new line.

RobbeSneyders commented on docs/documentation_guide.md:
```diff
@@ -0,0 +1,95 @@
+
+## Fondant Explorer
+
+Learn how to use the Fondant [data explorer](data_explorer.md) to explore the outputs of your
```
Also intermediate steps :slightly_smiling_face:

RobbeSneyders commented on docs/guides/installation.md:
````diff
@@ -0,0 +1,21 @@
+
+```
+pip install fondant
+```
````
I would document our extras here as well.

PhilippeMoussalli replied:
done!

RobbeSneyders commented on docs/runners/kfp_infrastructure.md:
I don't think this file is included in the tree. Not sure if it's needed though, it's linked from the kubeflow runner page and you can find it via search.

PhilippeMoussalli replied:
Added it to the advanced concepts

RobbeSneyders commented on docs/runners/kfp_infrastructure.md:
```diff
@@ -1,23 +1,31 @@
 !!! note "IMPORTANT"
-    - These script serve just a kickstart to help you setup Kubeflow for running Fondant, these are not production ready environments.
-    - Spinning up a cluster on a cloud vendor will incur a cost.
-    - You should never run a script without inspecting it so please familiarize yourself with the commands defined in the Makefiles and adapt it to your own needs.
+- These script serve just a kickstart to help you setup Kubeflow for running Fondant, these are not
+production ready environments.
+- Spinning up a cluster on a cloud vendor will incur a cost.
+- You should never run a script without inspecting it so please familiarize yourself with the
+commands defined in the Makefiles and adapt it to your own needs.
```
This breaks the admonition.

RobbeSneyders commented on docs/runners/kfp_infrastructure.md:
```diff
@@ -1,23 +1,31 @@
- 
-If you already have setup a kubernetes cluster and you have configured kubectl you can install kubeflow pipelines following this [guide](https://www.kubeflow.org/docs/components/pipelines/v1/installation/standalone-deployment/#deploying-kubeflow-pipelines)
 
+If you already have setup a kubernetes cluster and you have configured kubectl you can install
```
```suggestion
If you already have a kubernetes cluster  set up and you have configured kubectl you can install
```

RobbeSneyders reviewed (commented):
Thanks @PhilippeMoussalli! This is gonna be a great improvement!

Can you address my smaller comments in this PR and create tickets for the larger ones? Then we can already merge this PR and continue the work in smaller separated PRs.

I think we can still have a look at some individual pages and see how we can write them more from a user perspective and how they can use it, instead of explaining how it works. I'll run through the documentation again when it's merged and create tickets where I think this is the case.

Hakimovich99 commented on docs/guides/first_pipeline.md:
````diff
@@ -1,26 +1,9 @@
-
 Clone the Fondant GitHub repository
 
 ```
````
Link at the bottom of the page "[guide](guides/build_a_simple_pipeline.md)." down. Must send to "Building your own pipeline.

PhilippeMoussalli replied:
Good catch, thanks

Hakimovich99 reviewed (commented):
This top-bottom approach where we explain first how to build a pipeline before the components looks good to me. Very intuitive. 
I would add in the "documentation guide" under "getting started with Fondant" a bullet point to the guides  (“building your own pipeline” and “implement custom components”). I think it is important to have them. I have pinned in the docs where I think these should be added.
Also, in those guides, we should add more links to the "theory" which are the "Pipeline" and "Components" sections, but that's already a detail. 
I also wrote some other details I have found, mostly regarding broken links, but I haven't checked them all so some might remain. 
Thanks Philippe!!

Hakimovich99 commented on docs/documentation_guide.md:
```diff
@@ -0,0 +1,95 @@
+→ Get started by running your first fondant [pipeline](guides/first_pipeline.md) using the [local
+runner](runners/local.md). There you will learn how to use reusable components and how to build your
+own custom components.  
+→ Learn how to use the [data explorer](data_explorer.md) to explore the outputs of your pipeline.
```
(between the second and third bullet point)

PhilippeMoussalli replied:
Good catch, forgot about that one

PhilippeMoussalli commented on docs/caching.md:
```diff
@@ -0,0 +1,40 @@
+## Caching
```
Removed

PhilippeMoussalli commented:
@Hakimovich99 @RobbeSneyders 

Thanks for the feedback! I addressed the main points and added 2 tickets for other ones that should be tackled here #572

RobbeSneyders reviewed (approved):
Thanks @PhilippeMoussalli! Huge task, nice to see it done this quickly :)
